### PR TITLE
Code cleanup of Gforces module

### PR DESCRIPTION
**Pull request by jokoho48**
Code cleanup of Gforces module.

### CODE REVIEW

File: addons/gforces/functions/fnc_pfhUpdateGForces.sqf [hunk 1]
```diff
@@ -86,11 +80,12 @@ if (_suitCoef == 0) then {_suitCoef = 0.001};
 
 _gBlackOut = MAXVIRTUALG / _classCoef + MAXVIRTUALG / _suitCoef - MAXVIRTUALG;
 _gRedOut = MINVIRTUALG / _classCoef;
-
-["GForces", [], {format ["_g _gBO _coef: %1, %2, %3", _average, _gBlackOut, 2 * ((1.0 - ((_average - 0.30 * _gBlackOut) / (0.70 * _gBlackOut)) ^ 2) max 0) ]}] call EFUNC(common,log);
+#ifdef DEBUG_MODE_FULL
+    ["GForces", [], {format ["_g _gBO _coef: %1, %2, %3", _average, _gBlackOut, 2 * ((1.0 - ((_average - 0.30 * _gBlackOut) / (0.70 * _gBlackOut)) ^ 2) max 0) ]}] call EFUNC(common,log);
+#endif
```
Comment (jonpas) [inline addons/gforces/functions/fnc_pfhUpdateGForces.sqf]: Can `TRACE_1` be used here? Would need a bit of a refactor to the debug code probably.

Comment (jokoho48) [inline addons/gforces/functions/fnc_pfhUpdateGForces.sqf]: i think thats fine here

File: addons/gforces/functions/fnc_pfhUpdateGForces.sqf [hunk 2]
```diff
@@ -64,16 +55,19 @@ GVAR(oldVel) = _newVel;
...
+        true
+    } count GVAR(GForces);
+    _average = _sum / _sumDif;
 };
 
-_classCoef = ACE_player getVariable ["ACE_GForceCoef",
-    getNumber (configFile >> "CfgVehicles" >> (typeOf ACE_player) >> "ACE_GForceCoef")];
+_classCoef = ACE_player getVariable "ACE_GForceCoef";
+if (isNil "_classCoef") then {
+    _classCoef = getNumber (configFile >> "CfgVehicles" >> (typeOf ACE_player) >> "ACE_GForceCoef")];
```
Comment (PabstMirror) [inline addons/gforces/functions/fnc_pfhUpdateGForces.sqf]: `]` at end

File: addons/gforces/functions/fnc_pfhUpdateGForces.sqf [hunk 3]
```diff
@@ -13,26 +12,18 @@
...
 
-if (isNull ACE_player) exitWith {};
+ if !(alive ACE_player) exitWith {};
 
-if !(alive ACE_player) exitWith {};
+private ["_player", "_newVel", "_accel", "_currentGForce", "_average", "_sum", "_classCoef", "_suitCoef", "_gBlackOut", "_gRedOut", "_g", "_gBO", "_coef", "_strength"];
 
 GVAR(lastUpdateTime) = ACE_time;
-
-/*if !(vehicle ACE_player isKindOf "Air") exitWith {
```
Comment (PabstMirror) [inline addons/gforces/functions/fnc_pfhUpdateGForces.sqf]: If we are in the middle of a blackout, and eject from a air vehicle, the effect will never go away.

Comment (jokoho48) [inline addons/gforces/functions/fnc_pfhUpdateGForces.sqf]: ah now i understand yeah change it

File: addons/gforces/functions/fnc_pfhUpdateGForces.sqf [hunk 4]
```diff
@@ -64,16 +49,19 @@ GVAR(oldVel) = _newVel;
 */
 
 _average = 0;
-if (count GVAR(GForces) > 0) then {
+if (!isNil {GVAR(GForces) select 0}) then {
```
Comment (SilentSpike) [inline addons/gforces/functions/fnc_pfhUpdateGForces.sqf]: This is bad, I suggest `!(GVAR(GForces) isEqualTo [])`

Comment (jokoho48) [inline addons/gforces/functions/fnc_pfhUpdateGForces.sqf]: greate Idea

File: addons/gforces/functions/fnc_pfhUpdateGForces.sqf [hunk 5]
```diff
@@ -64,16 +49,19 @@ GVAR(oldVel) = _newVel;
...
-    _average = _sum / (count GVAR(GForces));
+        true
+    } count GVAR(GForces);
+    _average = _sum / _sumDif;
 };
 
-_classCoef = ACE_player getVariable ["ACE_GForceCoef",
-    getNumber (configFile >> "CfgVehicles" >> (typeOf ACE_player) >> "ACE_GForceCoef")];
+_classCoef = ACE_player getVariable "ACE_GForceCoef";
+if (isNil "_classCoef") then {
```
Comment (Glowbal) [inline addons/gforces/functions/fnc_pfhUpdateGForces.sqf]: I don't think this is cleaner as the previous code. As for performance increase; I expect we now got an if statement that is true more often then not. If anything, I'd expect this to be worse on average.

Comment (esteldunedain) [inline addons/gforces/functions/fnc_pfhUpdateGForces.sqf]: I agree.

File: addons/gforces/XEH_postInit.sqf [hunk 6]
```diff
@@ -9,7 +9,6 @@ GVAR(GForces_CC) ppEffectForceInNVG true;
 GVAR(GForces_CC) ppEffectAdjust [1,1,0,[0,0,0,1],[0,0,0,0],[1,1,1,1],[10,10,0,0,0,0.1,0.5]];
 GVAR(GForces_CC) ppEffectCommit 0.4;
 
-GVAR(lastUpdateTime) = 0;
 GVAR(oldVel) = [0,0,0];
 
-[FUNC(pfhUpdateGForces), 0, []] call CBA_fnc_addPerFrameHandler;
+[DFUNC(pfhUpdateGForces), INTERVAL, []] call CBA_fnc_addPerFrameHandler;
```
Comment (esteldunedain) [inline addons/gforces/XEH_postInit.sqf]: Although marginally more efficient, this will break the calculation when time is accelerated.

**Pull request closed without merging**